Assets/Mirage/Weaver/Processors/SyncVarProcessor.cs

```diff
@@ -488,6 +489,20 @@ void GenerateSerialization()
                 foreach (FoundSyncVar syncVar in behaviour.SyncVars)
                 {
                     WriteFromField(worker, helper.WriterParameter, syncVar);
+
+                    if (syncVar.FireOnServer && syncVar.HasHook)
+                    {
+                        VariableDefinition oldValue = helper.Method.AddLocal(syncVar.OriginalType);
+
+                        // call the hook
+                        // Generates: OnValueChanged(oldValue, this.syncVar)
+                        WriteCallHookMethodUsingField(worker, syncVar.Hook, oldValue, syncVar);
+                    }
+                    else if (syncVar.FireOnServer && !syncVar.HasHook)
+                    {
+                        Debug.LogWarning(
```

Comment:
Debug can't be used inside of Weaver because unity's internals are sometimes not loaded. You need to use the weaver logger instead. 

I think in this case throwing `SyncVarException`  or `HookMethodException` will work because it'll be caught and logged as an error inside `ProcessSyncVars`

Reply:
Changed it to use HookMethodException